packages/server/utils/AzureDevOpsServerManager.ts

```diff
@@ -442,6 +469,58 @@ class AzureDevOpsServerManager {
     return {error: undefined, projects: teamProjectReferences}
   }
 
+  async getProjectProperties(instanceId: string, projectId: string) {
+    let firstError: Error | undefined
+    const uri = `https://${instanceId}/_apis/projects/${projectId}/properties?keys=System.CurrentProcessTemplateId`
+    const result = await this.get<ProjectProperties>(uri)
+    if (result instanceof Error) {
+      if (!firstError) {
+        firstError = result
+      }
+    }
+    const requestedProperties = result as ProjectProperties
+    return {error: firstError, projectProperties: requestedProperties}
+  }
+
+  async getProjectProcessTemplate(instanceId: string, projectId: string) {
+    let firstError: Error | undefined
+    const result = await this.getProjectProperties(instanceId, projectId)
+    if (!!result.error) {
+      if (!firstError) {
+        firstError = result.error
+      }
+    }
+    const processTemplateProperty = result.projectProperties.value[0]
+    if (processTemplateProperty?.name !== 'System.CurrentProcessTemplateId') {
+      return {error: firstError, projectTemplate: ''}
+    }
+    const processTemplateDetailsResult = await this.getProcessTemplate(
+      instanceId,
+      processTemplateProperty?.value
+    )
+    if (!!processTemplateDetailsResult.error) {
+      if (!firstError) {
+        firstError = processTemplateDetailsResult.error
+      }
+    }
+    return {error: firstError, projectTemplate: processTemplateDetailsResult.process}
+  }
+
+  async getProcessTemplate(instanceId: string, processId: string) {
+    let firstError: Error | undefined
+    const uri = `https://${instanceId}/_apis/process/processes/${processId}?api-version=6.0`
+    const result = await this.get<Process>(uri)
+    if (result instanceof Error) {
+      if (result.message.includes('VS402362', 0)) {
```

Comment:
+1 what does `VS402362` refer to? Perhaps that could be assigned to a variable so it's more readable

Reply:
changed